identity-service/src/routes/instagram.js

```diff
@@ -1,77 +1,34 @@
-const Redis = require('ioredis')
 const request = require('request')
 const config = require('../config.js')
 const models = require('../models')
 const txRelay = require('../relay/txRelay')
-const redisClient = new Redis(config.get('redisPort'), config.get('redisHost'))
 
 const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError } = require('../apiHelpers')
 
-const INSTAGRAM_URL_POLLING_FREQ_MS = 500
 const getInstagramURL = (username) => {
   const instagramProfileUrl = config.get('instagramProfileUrl') || 'https://www.instagram.com/%USERNAME%/channel/?__a=1'
   return instagramProfileUrl.replace('%USERNAME%', username)
 }
-const getInstagramProfileRedisKey = (username) => {
-  return `instagramProfile:${username}`
-}
 
 /**
  * This file contains the instagram endpoints for oauth
  * For official documentation on the instagram oauth flow check out their site
  * https://www.instagram.com/developer/authentication/
  */
 module.exports = function (app) {
-  const instagramQueue = []
-  const doInstagramProfileFetching = () => {
-    setInterval(async () => {
-      if (instagramQueue.length > 0) {
-        const username = instagramQueue.shift()
-        try {
-          const res = await doRequest(getInstagramURL(username))
-          const json = JSON.parse(res)
-          if (json.graphql.user.full_name) {
-            await redisClient.set(getInstagramProfileRedisKey(username), res, 'EX', 60 * 60 * 24)
-          } else {
-            throw new Error(`Failed to fetch instagram profile for ${username}`)
-          }
-        } catch (e) {
-          await redisClient.set(getInstagramProfileRedisKey(username), 'error', 'EX', 60)
-        }
-      }
-    }, INSTAGRAM_URL_POLLING_FREQ_MS)
-  }
-  doInstagramProfileFetching()
-
   app.get('/instagram/profile', handleResponse(async (req, res, next) => {
     const username = req.query.username
-    const maxAttempts = req.query.maxAttempts || 100
+
     if (!username) {
       return errorResponseBadRequest('Missing username parameter')
     }
-    const redis = req.app.get('redis')
-
-    const key = getInstagramProfileRedisKey(username)
-    let value = await redis.get(key)
-    if (value === 'error') {
-      return errorResponseServerError(`Failed to fetch instagram profile for ${username}`)
-    }
-    if (!value) {
-      instagramQueue.push(username)
-      req.logger.info(`Instagram queue (${instagramQueue.length}): ${instagramQueue}`)
-    }
-    let attempts = 0
-    while (!value && attempts < maxAttempts) {
-      await new Promise(resolve => setTimeout(resolve, 500))
-      value = await redis.get(key)
-      attempts += 1
-    }
-
-    if (value) {
-      const json = JSON.parse(value)
+    try {
+      const res = await doRequest(getInstagramURL(username))
```

Comment:
do we not want any retries with this? what if the ig api is down for this one call?

Reply:
retries are handled upstream. the code being deleted here is removing retries / a polling mechanism to rate limit ourselves

Reply:
looool